ir/function.h

```diff
@@ -212,4 +214,23 @@ class DomTree final {
     void printDot(std::ostream &os) const;
 };
 
+class LoopAnalysis final {
+  Function &f;
+  CFG cfg;
+
+  std::map<const BasicBlock*, unsigned> number;
```

Comment:
do we need this stuff to be in the header? Most of this stuff seems to be private to the implementation that is run ocne and the clients of this class never lookup this information?

Reply:
Yes, all those three containers are used later in the analysis() and printDot() since DFST numbers are required to check ancestor relation between a pair of vertices. To get rid of those class fields I could either inline the getDFST() inside analysis(), or pass those containers as return values. What do you think @nunoplopes ?